logging/FileAndFolderHandler.cpp

```diff
@@ -0,0 +1,150 @@
+#include "FileAndFolderHandler.hpp"
+#include <sys/stat.h>
+#include <errno.h>
+#include <iostream>
+#include <fstream>
+
+namespace logging
+{
+    /**
+     * @brief Create the folder if it doesn't exist
+     *
+     * @param folder_name Name of the folder to create.
+     *
+     * @throw std::runtime_error if the folder can't be created
+     */
+    void FileAndFolderHandler::create_folder(std::string folder_name)
+    {
+        if (mkdir(folder_name.c_str(), 0777) != 0) {
+            switch(errno) {
```

Comment:
Don't use a switch like that, just throw a runtime_error with `errno` translated to a string using `strerror`, that is enough and more readable